lib/index.js

```diff
@@ -1,171 +1,100 @@
 'use strict';
 
 var async = require('async');
+var rarity = require('rarity');
 
-
-/**
- * Quickly build some fixtures, respecting creation constraints.
- * Example use :
- * var rawDataset = {
- *   company: {},
- *   user: {
- *     company: create.defer(dataset, 'company')
- *   },
- *   document: {
- *     company: create.defer(dataset, 'company'),
- *     filePath: '/tmp/somewhere,
- *     metadata: {
- *       title: 'AnyFetch document number 1',
- *     },
- *   }
- * };
- *
- * This will create a company, a user and a document and return them in dataset.
- */
-module.exports = function buildInitialState(rawDataset, dataset, waitForES, cb) {
-  if(!cb) {
-    cb = waitForES;
-    waitForES = false;
+module.exports = function createDataset(rawDataset, filledDataset, callback) {
+  var dataset;
+  if(!callback) {
+    dataset = {};
+    callback = filledDataset;
+  }
+  else {
+    dataset = filledDataset;
   }
 
-  /**
-   * Build each items in `objects` of type `model`, register them in dataset, then call cb.
-   */
-  var createObjects = function(model, objects, cb) {
-    var stack = [];
-    for(var key in objects) {
-      // Build data for this object
-      var data = {};
-      for(var subkey in objects[key]) {
-        if(objects[key][subkey] instanceof Function) {
-          data[subkey] = objects[key][subkey]();
-        }
-        else {
-          data[subkey] = objects[key][subkey];
+  // Function called by async.auto, see below
+  var generateObject = function(model, key, cb) {
+    module.exports.config[model].generator(rawDataset[key], function(err, result) {
```

Comment:
Isn't just moving the problem elsewhere? I will have to handle it line 74 anyway, right?